MaxText/train.py

```diff
@@ -282,10 +283,14 @@ def train_loop(config, state=None):
     last_step_completion = new_time
 
     if checkpoint_manager is not None:
-      if checkpoint_manager.save(step, state):
```

Comment:
this code is more complex and not any better than just writing the normal way -- sharding should be ~free.